### DISCUSSION TITLE
Offload calculation of views to workers?

### DISCUSSION THREAD

**Discussion by barbalex** [mobxjs/mobx-state-tree#2031] [Ideas]
**_Feature request_**

**Is your feature request related to a problem? Please describe.**

Calculating views synchronously can become a burden for the ui.
Seems to a noob like me it would be great if their calculation could be offloaded to workers.

**Describe the solution you'd like**

There is a closed issue to generally offload store calculations to workers (https://github.com/mobxjs/mobx-state-tree/issues/1318).

It was closed because though an interesting idea there was no clear way forward yet.
I am not adding anything substantial to that situation. So feel free to close this issue too.

But:
- Maybe things have changed, now that react 18 supports concurrency?
- Maybe it would help to focus only on views?

**Describe alternatives you've considered**

This is way out of my league. Above ideas may be way off.

**Are you willing to (attempt) a PR?**
* [ ] Yes
* [x] No

**Comment by coolsoftwaretyler** [comment]
Hey @barbalex! 

Sorry it took so long for anyone to get back to you. I think this is an interesting idea. But much like the issue you referenced, I also think that the recommendation/closing is appropriate here, as discussed in https://github.com/mobxjs/mobx-state-tree/issues/1318#issuecomment-511149401.

But what I'll do to make sure this conversation is available is convert it to a discussion. That'll close out the issue, but leave the opportunity for you and/or other folks to think about this more. If we get someone to champion an approach or implementation, maybe we can find a way forward.

**Comment by chakrihacker** [comment]
I want to add one thing here. Moving to WebWorkers should be optional. The reason is MST can be used outside of the web too, something like react native which doesn't have access to web workers
